real_spike/practice_acquire.py

```diff
@@ -0,0 +1,250 @@
+
+    sglx.c_sglx_close(hSglx)
+    sglx.c_sglx_destroyHandle(hSglx)
+
+
+def get_params(ip_address: str, port: int):
+    """Gets the most recent run parameters used for SpikeGLX."""
+    print("Get params...")
+    hSglx = sglx.c_sglx_createHandle()
+    ok = sglx.c_sglx_connect(hSglx, ip_address.encode(), port)
+
+    if ok:
+        print("Successfully connected to SpikeGLX")
+        nval = c_int()
+        len = c_int()
+        ok = sglx.c_sglx_getParams(byref(nval), hSglx)
+        if ok:
+            kv = {}
+            for i in range(0, nval.value):
+                line = sglx.c_sglx_getstr(byref(len), hSglx, i).decode()
+                parts = line.split('=')
+                kv.update({parts[0]: parts[1]})
+            print("{}".format(kv.items()))
+
+    if not ok:
+        print("error [{}]\n".format(sglx.c_sglx_getError(hSglx)))
+
+    sglx.c_sglx_close(hSglx)
+    sglx.c_sglx_destroyHandle(hSglx)
+
+
+def get_datadir(ip_address: str, port: int):
+    """Gets the global data directory."""
+    hSglx = sglx.c_sglx_createHandle()
+    ok = sglx.c_sglx_connect(hSglx, ip_address.encode(), port)
+
+    if ok:
+        data_dir = c_char_p()
+        ok = sglx.c_sglx_getDataDir(byref(data_dir), hSglx, 0)
+
+        if ok:
+            print(data_dir.value)
+        if not ok:
+            print("error [{}]\n".format(sglx.c_sglx_getError(hSglx)))
+
+            sglx.c_sglx_close(hSglx)
+            sglx.c_sglx_destroyHandle(hSglx)
+            return
+
+    if not ok:
+        print("error [{}]\n".format(sglx.c_sglx_getError(hSglx)))
+
+    sglx.c_sglx_close(hSglx)
+    sglx.c_sglx_destroyHandle(hSglx)
+
+
+def get_probes(ip_address: str, port: int):
+    """
+    Gets the list of available probe list. Should return in the format:
+    (probeID,nShanks,partNumber)
+    """
+    hSglx = sglx.c_sglx_createHandle()
+    ok = sglx.c_sglx_connect(hSglx, ip_address.encode(), port)
+
+    if ok:
+        list = c_char_p()
+        ok = sglx.c_sglx_getProbeList(byref(list), hSglx)
+        if ok:
+            #print(sglx.c_sglx_getstr(list))
+            print(list.value)
+    if not ok:
+        print("error [{}]\n".format(sglx.c_sglx_getError(hSglx)))
+
+    sglx.c_sglx_close(hSglx)
+    sglx.c_sglx_destroyHandle(hSglx)
+
+
+def get_imec_params(ip_address: str, port: int, ip=0):
+    """
+    Get parameters for a given imec probe. Probe id specified by `ip`.
+
+    To get params of all available probes, see c_sglx_getParamsImecCommon.
+    """
+    hSglx = sglx.c_sglx_createHandle()
+    ok = sglx.c_sglx_connect(hSglx, ip_address.encode(), port)
+
+    if ok:
+        nval = c_int()
+        ok = sglx.c_sglx_getParamsImecProbe(byref(nval), hSglx, ip)
+        if ok:
+            kv = {}
+            for i in range(0, nval.value):
+                line = sglx.c_sglx_getstr(byref(len), hSglx, i).decode()
+                parts = line.split('=')
+                kv.update({parts[0]: parts[1]})
+            print("{}".format(kv.items()))
```

Comment:
What does this do, can't you just `print(kv.items())` ? Or using `pprint` if you want to make it look nice